### PR TITLE
Added support for favorites, recents and links.

### PR DESCRIPTION
**Pull request by toddsouthenbentley**
This PR adds support for favorites and recent projects. 

I also enhanced the getByQuery function to return both the projects and the links. This function is now public. The links are functions that return more data and links. This allows for a paged user interface to be used with this client.

### CODE REVIEW

**Comment by toddsouthenbentley** [comment]
Closing so I can merge the latest code.

**Comment by toddsouthenbentley** [comment]
Reopening, can't seem to push changes to this branch.

File: src/ProjectsAccessProps.ts [hunk 1]
```diff
@@ -41,18 +41,51 @@ export interface ProjectsSearchArg {
...
+  { pagination?: ProjectsPaginationArg, search?: ProjectsSearchArg, source?: never };
+
 /** Methods for accessing projects
  * @beta
 */
 export interface ProjectsAccess {
   /** Get projects associated with the requester */
   getAll(accessToken: AccessToken, arg?: ProjectsQueryArg): Promise<Project[]>;
+  /** Get projects and links by query */
+  getByQuery(accessToken: AccessToken, queryArg?: ProjectsQueryArg): Promise<ProjectsQueryResult>;
```
Comment (calebmshafer) [inline src/ProjectsAccessProps.ts:68]: Do we need to update the main interface? This is technically a breaking change to the API by enforcing a new method to be implemented.

Comment (calebmshafer) [reply src/ProjectsAccessProps.ts:68]: Also, can you align with the `getAll` signature and use `arg` instead of `queryArg`?

Comment (toddsouthenbentley) [reply src/ProjectsAccessProps.ts]: OK, I removed the new method in the interface and changed the signature of the getByQuery function.

**Review by calebmshafer — APPROVED** [review]
LGTM

**Pull request merged**